### PR TITLE
CI: Use new documentation previewer

### LINKED ISSUE #50832: DOC: Preview rendered documentation of PRs
The idea is that when someone opens a PR 12345, after the CI finishes building the docs, we have a url https://pandas.pydata.org/docs/dev/pr/12345/ (we can probably think of a better url) to preview how are the docs with the pr changes.

We discussed this long ago, and couldn't find an easy way to implement it with the CI. The main challenge is that the CI of a PR doesn't have access to secrets, so we can't publish to a server in the same way we do for commits to main. But we've got options, like to add a webhook to the CI, and implement a small service in a server, that when it receives a request from a webhook, it finds the artifact for that job, downloads and decompresses it to a directory, which is served as a static http server to the desired url.

We will need a job to clean up docs after X days, since each copy of our web/docs is around 350M, but that should be easy.

In the future we can even try to detect where the changes happened, and add a comment to the PR with the links to the pages of interest. Not sure how difficult would that be for docstrings, but for regular rst files (e.g. the user guide) it should be quite easy.

Please let me know if there are ideas, or any objection. Otherwise I'll give this a try in few days.

### PR DESCRIPTION
**Pull request by datapythonista**
- [X] closes #50832

We've got a documentation previewer service for few months now, but the whole infrastructure powering it was just a PoC not very suitable for the long term (it didn't have error control, was running in a screen session, the workflow wasn't optimal, a github token was hardcoded, it required to remove old previews manually...).

Now we've got a service that IMHO is ready for production. Everything is built independently from pandas in this repo:
https://github.com/pandas-dev/github-doc-previewer/. In the pandas side the only thing we need is to call a GitHub action (check the diff in this PR).

For most people there won't be a significant difference between the previous PoC and the new system, as the UX is the same. Besides stability the main differences are:
- A preview will be generated only for the PRs where it's requested (with the `/preview` comment). Previously we published the previews of every PR.
- The new service is pandas-agnostic. If there is interest, any other repo can use it, the only thing needed is that the GitHub organization of the project is added to a setting in the server. The whole `pandas-dev` organization is of course added. Not sure if we have any other repo where this can be useful, but only adding the GitHub action is enough to have previews enabled.
- The previous point has a small implication. The urls of the previews now contain the org and repo. So, instead of pandas.pydata.org/preview/1234/ the new url is now pandas.pydata.org/preview/pandas-dev/pandas/1234/

Any feedback or problem please let me know.

### CODE REVIEW

**Comment by datapythonista** [comment]
I'll be merging this soon if there are no objections, so we can use the previewer again (with the new service).

File: .github/workflows/comment-commands.yml [hunk 1]
```diff
@@ -24,12 +24,10 @@ jobs:
     concurrency:
       group: ${{ github.actor }}-preview-docs
     steps:
-      - run: |
-          if curl --output /dev/null --silent --head --fail "https://pandas.pydata.org/preview/${{ github.event.issue.number }}/"; then
-            curl -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" -d '{"body": "Website preview of this PR available at: https://pandas.pydata.org/preview/${{ github.event.issue.number }}/"}' https://api.github.com/repos/${{ github.repository }}/issues/${{ github.event.issue.number }}/comments
-          else
-            curl -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" -d '{"body": "No preview found for PR #${{ github.event.issue.number }}. Did the docs build complete?"}' https://api.github.com/repos/${{ github.repository }}/issues/${{ github.event.issue.number }}/comments
-          fi
+      - uses: pandas-dev/github-doc-previewer@master
```
Comment (mroeschke) [inline .github/workflows/comment-commands.yml:27]: In the future it would be great if we could use a tag for stability but not a big issue

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks all your hard work here @datapythonista

**Comment by datapythonista** [comment]
I had to increase the timeout on nginx, since the requests stays life for a bit more than one minute until the preview is ready and the comment can be posted, but now everything seems to be working fine with the new version.

File: .github/workflows/comment-commands.yml (hunk 1 above)
Comment (datapythonista) [reply .github/workflows/comment-commands.yml:27]: Good point. At least for now I think all releases of the doc previewer will be done for pandas, but I created #57190 anyway, so we can decide when we upgrade.

**Pull request merged**